### ISSUE TITLE
KeyError: 'known_reaction'

### ISSUE THREAD

**Issue by niraito** [EPFL-LCSB/ARBRE#1]
Hello,

I was trying to run the code as following,

```
python3 main.py norcoclaurine_precursorsList
```
However I receive the following error:

```
Traceback (most recent call last):
  File "/home/nilay/anaconda3/envs/rdkit/lib/python3.6/site-packages/pandas/core/indexes/base.py", line 2898, in get_loc
    return self._engine.get_loc(casted_key)
  File "pandas/_libs/index.pyx", line 70, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 101, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 1675, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 1683, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: 'known_reaction'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "main.py", line 13, in <module>
    dat = Data()
  File "/mnt/c/Users/nilay/Documents/phd/pathdeneme/lscb-arbre/ARBRE/code/data.py", line 70, in __init__
    self.known_steps_all = self.df_network[self.df_network['known_reaction']==1]['UID of pair'].to_list()
  File "/home/nilay/anaconda3/envs/rdkit/lib/python3.6/site-packages/pandas/core/frame.py", line 2906, in __getitem__
    indexer = self.columns.get_loc(key)
  File "/home/nilay/anaconda3/envs/rdkit/lib/python3.6/site-packages/pandas/core/indexes/base.py", line 2900, in get_loc
    raise KeyError(key) from err
KeyError: 'known_reaction'
```

Where does the `known_reaction` come from? And what might be the problem here?

**Comment by anasves** [comment]
Hello,

'known_reaction' is a name of the column in data/network.csv. If this file is empty, then the KeyError will appear. Please, make sure that you used git LFS to correctly download the files. Alternatively, download them directly from this git and put into data folder.

Please, let me know if the problem is solved.

**Comment by niraito** [comment]
Hello,

Thank you for your reply. I downloaded the files by using git LFS, and it worked well. However, the same code `python3 main.py norcoclaurine_precursorsList` stuck for a long time as follows:

```
filter out the reactions that are in excludelists/reactions.txt
1086224  balanced reactions selected out of  1086224 initial reactions
Load network...
Size of the network loaded:  64203
Searching for pathways
Number of defined source metabolites in the network: 3
```

Then, it continued and resulted in the following error:

```
Time to find all pathways:  3402.7435245513916
time to find raw pathways: 10428.153435707092
Total number of reactions in all pathways:  69
Traceback (most recent call last):
  File "main.py", line 68, in <module>
    vis.drawPathways()
  File "/mnt/c/Users/nilay/Documents/phd/pathdeneme/lscb-arbre/ARBRE/code/visualize.py", line 75, in drawPathways
    self.drawPathwayGraph(row)
  File "/mnt/c/Users/nilay/Documents/phd/pathdeneme/lscb-arbre/ARBRE/code/visualize.py", line 128, in drawPathwayGraph
    nx.draw_networkx_edges(G, pos=positions, width=weights, edge_color=colors)
  File "/home/nilay/anaconda3/envs/rdkit/lib/python3.6/site-packages/networkx/drawing/nx_pylab.py", line 563, in draw_networkx_edges
    if not cb.iterable(width):
AttributeError: module 'matplotlib.cbook' has no attribute 'iterable'
```

I have received this `AttributeError` with some of the other sample codes. Even so, I have results in the output file. However, for instance, I get a 5-steps pathway for the scoulerine case study, while it is a 7-steps pathway on the paper. Does AttributeError affect it?

My system has the requirements with the following versions;
* networkx==2.2
* numpy==1.19.5
* matplotlib==3.3.4
* seaborn==0.11.2
* pandas==1.1.5
* pillow==6.2.1

If I need to upgrade or downgrade, what are the version specifications?

**Comment by anasves** [comment]
Hello,
3402 seconds to find pathways  is far more than is expected. For me it takes several seconds (MacOS 10.14, 2.7 GHz 12-Core Intel Xeon E5, 32GB of memory). It would be nice if you specified your processor/memory so that we knew the time estimate depending on the system.
Please, see this Stack Overflow issue solving the Attribute Error:
https://stackoverflow.com/questions/63198347/attributeerror-module-matplotlib-cbook-has-no-attribute-iterable
You probably need to downgrade your matplotlib version to 2.2.3, otherwise according to that issue it is apparently incompatible with networkx.
Also, I have updated this git version with the parameter that was used in the paper (not atom-mapped reactant-product pairs were excluded in that case). Please, pull the latest version and let me know if you get the 7-steps pathway.

**Comment by niraito** [comment]
Hello again,

Sorry for my inconvenient complaints 🙈 I was just assuming the case studies would be small and can be done quickly, on the other hand, my system specifications are a bit shabby I suppose: 

Processor	Intel(R) Core(TM) i5-1035G1 CPU @ 1.00GHz   1.19 GHz
Installed RAM	12.0 GB (11.8 GB usable)
System type	64-bit operating system, x64-based processor

Edition	Windows 11 Home
Version	21H2
OS build	22000.613

I have solved AttributeError by downgrading matplotlib version to 2.2.3, as you suggested!

Also, I have pulled the latest git version and now I can get the 7 steps pathway for scoulerine, except for some slight differences in EC numbers of pathway enzymes. 

Thank you very much!
